apps/editing-toolkit/editing-toolkit-plugin/wpcom-global-styles/index.php

```diff
@@ -0,0 +1,83 @@
+<?php
+/**
+ * Limit Global Styles on WP.com to paid plans.
+ *
+ * @package full-site-editing-plugin
+ */
+
+/**
+ * Checks if Global Styles should be limited on the given site.
+ *
+ * @param  int $blog_id Blog ID.
+ * @return bool Whether Global Styles are limited.
+ */
+function wpcom_should_limit_global_styles( $blog_id = 0 ) {
+	if ( ! $blog_id ) {
+		$blog_id = get_current_blog_id();
+	}
+
+	// Do not limit Global Styles on Atomic sites for now, because blog stickers are not exposed
+	// to these sites and the project is still in a development stage that requires sites to have
+	// a certain blog sticker before restricting the feature. This is a temporary check that will
+	// be removed as part of the public launch.
+	if ( ! defined( 'IS_WPCOM' ) || ! IS_WPCOM ) {
+		return false;
+	}
+
+	// Do not limit Global Styles on sites created before we made it a paid feature. This cutoff
+	// blog ID needs to be updated as part of the public launch.
+	if ( $blog_id < 210494207 ) {
+		return false;
+	}
+
+	if ( wpcom_site_has_feature( WPCOM_Features::GLOBAL_STYLES, $blog_id ) ) {
+		return false;
+	}
+
+	// During the development stage, we only limit Global Styles on sites that have opted in. This
+	// is a temporary check that will be removed as part of the public launch.
+	return has_blog_sticker( 'wpcom-limit-global-styles', $blog_id );
+}
+
+/**
+ * Customizes the Global Styles feature in the site editor on sites missing the required paid plan.
+ *
+ * @return void
+ */
+function wpcom_maybe_limit_site_editor_global_styles() {
```

Comment:
Besides, one of our old team leads would have scolded you for using the `maybe` word. 😄

Reply:
> In fact, this function simply adds the JS and CSS for the GS Gating modal!
>
> The name is very misleading, as it suggests that this function is actually limiting GS in the site editor.

Fair enough! I don't like a generic name like `wpcom_global_styles_enqueue_scripts_and_styles` because that doesn't add any context to the developer (forcing them to read the rest of the code to understand what it does), so I'll probably rename it to something like `wpcom_site_editor_global_styles_add_paid_feature_alerts` which should be more accurate.

> Besides, one of our old team leads would have scolded you for using the `maybe` word. 😄

Blame @obenland! He fooled me to like all these WordPress conventions and now I cannot stop using them.

Reply:
> I don't like a generic name like wpcom_global_styles_enqueue_scripts_and_styles because that doesn't add any context to the developer

IMHO it's the generic name here is enough, since that's exactly what the function does.
I wouldn't expect folks to look at the enqueue function to understand what's going on, but maybe that's me, biased from hundreds of generic enqueue functions. 🙂

Plus, I realize now I forgot to add a further request that would have clarified my comment: a readme for the new package.
IMHO it's super important, especially in a monorepo context, to centralize the documentation, so that folks wouldn't need to look into the code to learn what this package does.

> Blame @obenland!

You know I can't!
But now I'm tempted to ping the helping cat to see what she has to say about your maybes! 😄

Reply:
> I wouldn't expect folks to look at the enqueue function to understand what's going on

That's true, especially since we might change what the script does in the future without realizing that we should change the name of the function. So I ended up using the generic name

> a readme for the new package

Good call! I just added a very basic one in https://github.com/Automattic/wp-calypso/pull/67748/commits/a69978538687c3ea6cb45603cb835eabfa6f0db4, but let me know if you think it should be improved.